### PR TITLE
fix: add offset and limit to key transfer API

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] Enhancement

## Description

Add offset and limit to key transfer API

## Motivation and Context

This adds `from` and `limit` parameters to Key transfer API and thus providing a way to transfer the keys in batches.

## How did you test it?

1. Run `/accounts/transfer` with `from` and `limit` parameters. It should respond with the `total_transferred` same as `limit`.

```bash
curl --location 'http://localhost:8080/accounts/transfer' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "from": 0,
    "limit": 2
}'
```

2. Do the same thing for `user/key/transfer`
```bash
curl --location 'http://localhost:8080/user/key/transfer' \
--header 'Content-Type: application/json' \
--header 'api-key: test_admin' \
--data '{
    "from": 0,
    "limit": 2
}'
````

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

File: crates/router/src/db/merchant_key_store.rs [hunk 1]
```diff
@@ -172,12 +174,16 @@ impl MerchantKeyStoreInterface for Store {
...
         let fetch_func = || async {
-            diesel_models::merchant_key_store::MerchantKeyStore::list_all_key_stores(&conn)
-                .await
-                .map_err(|err| report!(errors::StorageError::from(err)))
+            diesel_models::merchant_key_store::MerchantKeyStore::list_all_key_stores(
+                &conn, from, to,
+            )
+            .await
+            .map_err(|err| report!(errors::StorageError::from(err)))
         };
```
Comment (NishantJoshi00) [inline crates/router/src/db/merchant_key_store.rs:187]: Why is this a function? I doesn't seem to be doing anything concurrently

**Review by NishantJoshi00 — CHANGES_REQUESTED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by NishantJoshi00 — DISMISSED** [review]

File: crates/api_models/src/events.rs [hunk 2]
```diff
@@ -76,6 +76,7 @@ impl_misc_api_event_type!(
     BusinessProfileUpdate,
     BusinessProfileCreate,
     RevokeApiKeyResponse,
+    MerchantKeyTransferRequest,
```
Comment (lsampras) [inline crates/api_models/src/events.rs:79]: Can we add a new evnt type for this instead of miscellaneous

**Review by ThisIsMani — APPROVED** [review]

**Review by lsampras — DISMISSED** [review]

**Review by NishantJoshi00 — DISMISSED** [review]

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

File: crates/diesel_models/src/query/merchant_key_store.rs [hunk 3]
```diff
@@ -55,7 +55,11 @@ impl MerchantKeyStore {
         .await
     }
 
-    pub async fn list_all_key_stores(conn: &PgPooledConn) -> StorageResult<Vec<Self>> {
+    pub async fn list_all_key_stores(
+        conn: &PgPooledConn,
+        from: u32,
+        to: u32,
```
Comment (vspecky) [inline crates/diesel_models/src/query/merchant_key_store.rs:61]: The naming convention seems to be `from/limit` everywhere else, but is `from/to` here. Is this correct?

**Review by vspecky — APPROVED** [review]
LGTM

File: crates/diesel_models/src/query/merchant_key_store.rs (hunk 3 above)
Comment (dracarys18) [reply crates/diesel_models/src/query/merchant_key_store.rs:61]: Yes

Comment (dracarys18) [reply crates/diesel_models/src/query/merchant_key_store.rs]: Changed this

**Review by lsampras — APPROVED** [review]

**Review by vspecky — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

**Review by ThisIsMani — APPROVED** [review]

**Review by NishantJoshi00 — APPROVED** [review]

**Pull request merged**